src/systemctl/systemctl.c

```diff
@@ -6693,18 +6695,21 @@ static int systemctl_parse_argv(int argc, char *argv[]) {
                                 if (!arg_properties)
                                         return log_oom();
                         } else {
-                                const char *word, *state;
-                                size_t size;
+                                p = optarg;
+                                for(;;) {
+                                        _cleanup_free_ char *prop = NULL;
 
-                                FOREACH_WORD_SEPARATOR(word, size, optarg, ",", state) {
-                                        char *prop;
+                                        r = extract_first_word(&p, &prop, ",", 0);
+                                        if (r < 0)
+                                                return log_error_errno(r, "Failed to parse property: %s", optarg);
 
-                                        prop = strndup(word, size);
-                                        if (!prop)
-                                                return log_oom();
+                                        if (r == 0)
+                                                break;
 
-                                        if (strv_consume(&arg_properties, prop) < 0)
+                                        if (strv_push(&arg_properties, prop) < 0)
```

Comment:
else is not required

````
  if (isempty(optarg) && !arg_properties)
                                return log_error_errno(r, "--type requires arguments.");
 ````


yes strv_split uses twice .
1. to find how many to allocate
2. then actually it stores into them

which will look same as strv_split_extract if we replace with extract_first_word
except ExtractFlags